apps/api/src/app/widgets/e2e/get-subscriber-preference.e2e.ts

```diff
@@ -0,0 +1,61 @@
+import { NotificationTemplateEntity, SubscriberRepository, SubscriberPreferenceRepository } from '@novu/dal';
+import { UserSession } from '@novu/testing';
+import axios from 'axios';
+import { expect } from 'chai';
+
+describe('GET /widget/notifications/feed', function () {
+  let template: NotificationTemplateEntity;
+  let session: UserSession;
+  let subscriberToken: string;
+  let subscriberId: string;
+  let subscriberProfile: {
+    _id: string;
+  } = null;
+
+  beforeEach(async () => {
+    session = new UserSession();
+    await session.initialize();
+    subscriberId = SubscriberRepository.createObjectId();
+
+    template = await session.createTemplate({
+      noFeedId: true,
+    });
+
+    const { body } = await session.testAgent
+      .post('/v1/widgets/session/initialize')
+      .send({
+        applicationIdentifier: session.environment.identifier,
+        subscriberId,
+        firstName: 'Test',
+        lastName: 'User',
+        email: 'test@example.com',
+      })
+      .expect(201);
+
+    const { token, profile } = body.data;
+
+    subscriberToken = token;
+    subscriberProfile = profile;
+  });
+
+  it('should fetch a default user preference', async function () {
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+
+    await session.awaitRunningJobs(template._id);
+
+    const response = await axios.get(`http://localhost:${process.env.PORT}/v1/widgets/subscriber-preference`, {
+      headers: {
+        Authorization: `Bearer ${subscriberToken}`,
+      },
+    });
```

Comment:
Do you think we can abstract the session initialize and the widget requests under the session class? I see we have quite a lot of those usages already. Maybe doing something moregeneric to abstract the subscriber creation and initialization would be really nice.

Reply:
Love it! 
Done! ♥️